[Feat] location API + 드롭다운 리스트 설정

null4848 opened the pull request:
**Check List :memo:**
- [x] Pull Request 제목 : [유형] 변경 사항 요약
- [x] 설명 & 이슈 번호 작성

**설명 (필수)**

이 PR에서 어떤 변경을 했는지 상세히 설명해주세요.
관련된 Issue 번호를 포함하세요.

1. #20 

2. 변경 사항

Places API(New) 추가

이미 등록된 장소(POI) 클릭 이벤트를 통해서 정보 받아오기
poi - name,latlng,placeID

poi placeID 이용해서 타입, 정확한 주소 받아옴
-> placeID 오류가 나는 장소들이 있음 (이유를 모르겠음 - 나중에 수정 예정)

MapScreen 위치 정보 박스에 받아온 정보 뜨게 구현함
로그에 palce의 이름, 위치, 주소, 장소 유형 뜨게 구현함

MapViewModel -> 찜하기, 방문하기, 기록하기 API 호출함
공통 부분 handleApiResponse로 다룸
retrofitClient -> 각각 API 서비스 객체 초기화 (하나로 통합해서 사용할 가능성 있음)
MapScreen에서 호출함
-> 색상 변경 함수 이용해서 관리
-> 타입 겹치는 단어 제외하고 첫번째 타입만 넘기는 걸로 수정
각각 박스 버튼에 API호출함

TripsDropDownResponse 이용해서 title, tripId 다룸
dropdownAPI 이용해서 title+tripId 리스트로 가져옴
드롭다운 메뉴 아이템에 item.title로 가져옴
전체 여행 추출해서 먼저 배치하게끔 함


3. 스크린샷 (선택)
![image](https://github.com/user-attachments/assets/d0145eeb-a1bd-46b9-b102-0458eb9e661e)
![Screenshot_20250225_231554](https://github.com/user-attachments/assets/564b56bd-041f-4eba-b385-4779620719bc)

<br>

---

**review 참고 사항 (선택)**

추가로 알릴 내용이 있다면 여기에 작성해주세요.

1. 로그, UI 이미지 두 개 첨부했어요

2.  아직 찜, 방문, 기록  API 테스트는 아직 안됨 -> 백엔드 수정되면 바로 확인

null4848 committed:
feat: Places API 사용

Places API 이용해서 장소 주소, 이름, 타입 불러오는 코드

좌표 이용해서 주소 변환
주소를 이용해서 PlaceID 찾아오기
PlaceID를 이용해서 주소, 이름, 타입 가져오기

API 오류가 계속 뜸
일단 중간 저장용으로 push

null4848 committed:
feat: 장소 정보 받아오고 로그로 출력

null4848 committed:
feat: POI 이용해서 정보 받아오기

이미 등록된 장소(POI) 클릭 이벤트를 통해서 정보 받아오기
 poi - name,latlng,placeID
poi placeID 이용해서 타입, 정확한 주소 받아옴
-> placeID 오류가 나는 장소들이 있음
-> 주소가 짤려있는 건 구글에도 주소가 그렇게 뜸

moon0727 reviewed (approved):
현재 main과 충돌 있습니다! 해결 부탁드려요

moon0727 reviewed (changes requested):
현재 main과 충돌 있습니다! 해결 부탁드려요

null4848 committed:
Merge branch 'main' into 20-feat-location-api

null4848 committed:
feat: 의존성 추가

null4848 committed:
feat: mapPin API 연결 -> 코드만 짜둠

MapPinFavored Request. Response 추가

MapPinVisit Request, Response 추가

MapPinRecord Resquest, Response 추가

MapPinApiService -> favoredPin, visitPin, recordPin 정의

MapViewModel -> 찜하기, 방문하기, 기록하기 API 호출함
공통 부분 handleApiResponse로 다룸
retrofitClient -> 각각 API 서비스 객체 초기화 (하나로 통합해서 사용할 가능성 있음)
MapScreen에서 호출함
-> 색상 변경 함수 이용해서 관리
-> 타입 겹치는 단어 제외하고 첫번째 타입만 넘기는 걸로 수정
각각 박스 버튼에 API호출함
아직 API 실행?은 안함 코드만 짬

moon0727 committed:
fix : RetrofitClient 수정

- RetrofitClient 수정 -> MainActivity, GoogleAuthViewModel 수정
- TravelApiService 생성
- viewmodel package 생성

null4848 committed:
feat: mapPin 추가

null4848 committed:
Merge remote-tracking branch 'origin/31-feat-create-trip-api' into 20-feat-location-api

null4848 committed:
feat: API 연결

null4848 committed:
feat: 드롭다운 구현

TripsDropDownResponse 이용해서 title, tripId 다룸

dropdownAPI 이용해서
리스트 불러옴
전체 여행 추출해서 먼저 배치하게끔 함
드롭다운 메뉴 아이템에 item.title로 가져옴

null4848 committed:
feat: Api 삭제
